api/v1alpha1/stage_types.go

```diff
@@ -248,11 +248,16 @@ type GitRepoUpdate struct {
 type PullRequestPromotionMechanism struct {
 	// GitHub indicates git provider is GitHub
 	GitHub *GitHubPullRequest `json:"github,omitempty" protobuf:"bytes,1,opt,name=github"`
+	// GitHub indicates git provider is Gitlab
+	Gitlab *GitlabPullRequest `json:"gitlab,omitempty" protobuf:"bytes,1,opt,name=gitlab"`
```

Comment:
Nit:

GitLab, like GitHub, uses Pascal Case in their name. For consistency, can we rename internal datastructure names to `GitLab`? e.g.:
* `GitLabPullRequest`
* `GitLabProvider`
* `GitLabClient`

The json field name and protobuf name can still remain lowercase